web/docs/auth/overview.md

```diff
@@ -2,26 +2,631 @@
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+
+page AccountPage {
+  component: import Account from "@client/pages/Account.jsx",
+  authRequired: true
+}
+```
+
+```jsx title="client/pages/Account.jsx"
+import Button from './Button';
+import logout from '@wasp/auth/logout';
+
+const AccountPage = ({ user }) => {
+  return (
+    <div>
+      <Button onClick={logout}>Logout</Button>
+      {JSON.stringify(user, null, 2)}
+    </div>
+  );
+};
+
+export default AccountPage;
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+
+page AccountPage {
+  component: import Account from "@client/pages/Account.tsx",
+  authRequired: true
+}
+```
+
+```tsx title="client/pages/Account.tsx"
+import type { User } from '@wasp/entities';
+import Button from './Button';
+import logout from '@wasp/auth/logout';
+
+const AccountPage = ({ user }: { user: User }) => {
+  return (
+    <div>
+      <Button onClick={logout}>Logout</Button>
+      {JSON.stringify(user, null, 2)}
+    </div>
+  );
+};
+
+export default AccountPage;
+```
+</TabItem>
+</Tabs>
+
+#### Using the `useAuth` hook
+
+Wasp provides a React hook you can use in the client components - `useAuth`.
+
+This hook is a thin wrapper over Wasp's `useQuery` hook and returns data in the same format.
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="src/client/pages/MainPage.jsx"
+import useAuth from '@wasp/auth/useAuth'
+import { Link } from 'react-router-dom'
+import logout from '@wasp/auth/logout'
+import Todo from '../Todo'
+
+export function Main() {
+  const { data: user } = useAuth()
+
+  if (!user) {
+    return (
+      <span>
+        Please <Link to='/login'>login</Link> or <Link to='/signup'>sign up</Link>.
+      </span>
+    )
+  } else {
+    return (
+      <>
+        <button onClick={logout}>Logout</button>
+        <Todo />
+      < />
+    )
+  }
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="src/client/pages/MainPage.tsx"
+import useAuth from '@wasp/auth/useAuth'
+import { Link } from 'react-router-dom'
+import logout from '@wasp/auth/logout'
+import Todo from '../Todo'
+
+export function Main() {
+  const { data: user } = useAuth()
+
+  if (!user) {
+    return (
+      <span>
+        Please <Link to='/login'>login</Link> or <Link to='/signup'>sign up</Link>.
+      </span>
+    )
+  } else {
+    return (
+      <>
+        <button onClick={logout}>Logout</button>
+        <Todo />
+      < />
+    )
+  }
+}
+```
+</TabItem>
+</Tabs>
+
+### On the server
+
+#### Using the `context.user`
+
+When authentication is enabled, all [queries and actions](/docs/database/operations) will have access to the `user` through the `context` argument. `context.user` will contain all the fields from the user entity except for the password.
```

Comment:
Personal suggestion: it's best to avoid future tenses when possible.
```suggestion
When authentication is enabled, all [queries and actions](/docs/database/operations) have access to the `user` object through the `context` argument. `context.user` contains all User entity's fields, except for the password.
```